src/lsp/sql_preproc/sql_typeck.ml

```diff
@@ -0,0 +1,245 @@
+(**************************************************************************)
+(*                                                                        *)
+(*  Copyright (c) 2021-2023 OCamlPro SAS                                  *)
+(*                                                                        *)
+(*  All rights reserved.                                                  *)
+(*  This file is distributed under the terms of the                       *)
+(*  OCAMLPRO-NON-COMMERCIAL license.                                      *)
+(*                                                                        *)
+(**************************************************************************)
+
+open Cobol_data.Types
+
+let get_x_info (cu : Cobol_unit.Types.cobol_unit) name_str =
+  (* TODO: this needs to include refs from EXEC SQL INCLUDE stmt : TSQL042A *)
+  (* May raise Not_found | Cobol_unit.Qualmap.Ambiguous _ *)
+  Cobol_unit.Qualmap.find
+    (Cobol_unit.Qual.name
+       (Cobol_common.Srcloc.flagit name_str Cobol_common.Srcloc.dummy) )
+    cu.unit_data.data_items.named
+
+let get_length cu name =
+  try
+    let x_info = get_x_info cu name in
+    match x_info with
+    | Data_field { def = { payload = { field_size; field_layout; _ }; _ }; _ } ->
+        let size = Cobol_data.Memory.(as_bits field_size / 8) in
```

Comment:
Note: there may be some unwanted rounding error hidden here (number of bits here might not be a multiple of 8). Correct code would be:
```suggestion
        let size = (Cobol_data.Memory.as_bits field_size + 7) / 8 in
```

Also `Cobol_data.Memory.NOT_SCALAR` needs to be caught here.